packages/next/src/server/lib/router-server.ts

```diff
@@ -576,6 +577,21 @@ export async function initialize(opts: {
         )
       }
 
+      // For not found static assets, return plain text 404 instead of
+      // full HTML 404 pages to save bandwidth.
+      if (
+        parsedUrl.pathname &&
```

Comment:
when do we need to check `parsedUrl.pathname`? I thought it's always a valid string